src/TSHelper.ts

```diff
@@ -60,6 +60,14 @@ export function isAssignmentPattern(node: ts.Node): node is ts.AssignmentPattern
     return ts.isObjectLiteralExpression(node) || ts.isArrayLiteralExpression(node);
 }
 
+export function isDestructuringAssignment(node: ts.Node): node is ts.DestructuringAssignment {
```

Comment:
Typescript has this one too, any reason we're not using that?

Reply:
Are you sure about that? I couldn't find any functions that have a `is ts.DestructuringAssignment` type guard

Reply:
https://github.com/microsoft/TypeScript/blob/master/src/compiler/utilities.ts#L4023

Reply:
It's in the block with `@internal` annotation, so can't be used directly